substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -209,62 +228,111 @@ private int deflateBytes(long addr, byte[] b, int off, int len, int flush) {
         }
     }
 
+    // (JDK 11)src/java.base/share/native/libzip/Deflater.c-static jlong doDeflate(JNIEnv *env, jobject this, jlong addr,
+    //                       jbyte *input, jint inputLen,
+    //                       jbyte *output, jint outputLen,
+    //                       jint flush, jint params)
+    private static long doDeflate(long addr, CCharPointer input, int inputLen, CCharPointer output, int outputLen, int flush, int params) {
+        z_stream strm = WordFactory.pointer(addr);
+        int inputUsed, outputUsed;
+        int finished = 0;
+        int setParams = params & 1;
+
+        strm.set_next_in(input);
+        strm.set_next_out(output);
+        strm.set_avail_in(inputLen);
+        strm.set_avail_out(outputLen);
+
+        if (setParams != 0) {
```

Comment:
For converting a C `int` to a Java `boolean`, please use `CTypeConversion.toBoolean(int)`.

Reply:
It's a C int intentionally because of its role in the bit shift at the bottom of the method (this is mirroring the implementation code).  So I think it's better to keep it as-is: a value of 0 explicitly means do not set params, whereas a value of 1 means set params.  In this case I could use that conversion and it would happen to work because the domain of values is 0 or 1, but it's not really a boolean, it's a one-bit integer.

Reply:
You are right.